File: 프로그래머스/Level 1/하샤드 수.cpp
//https://programmers.co.kr/learn/courses/30/lessons/12947

#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int tmp = 0;
    int store = x;
    while(x > 0){
        tmp = tmp + x % 10;
        x /= 10;
    }
    if(store % tmp != 0) answer = false;
    return answer;
}

File: 프로그래머스/Level 1/하샤드 수.py
//https://programmers.co.kr/learn/courses/30/lessons/12947?language=python3

c++스럽게
def solution(x):
    answer = True

    tmp = x
    sum = 0
    while(tmp>0):
        sum = sum + tmp%10
        tmp = int(tmp/ 10)
    if(x % sum != 0 ):
        return False
    return answer
    
python스럽게
def solution(x):
    
    a = str(x)
    a = list(map(int,list(a)))
    return x % sum(a) == 0
